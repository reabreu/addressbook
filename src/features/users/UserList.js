import React from "react";
import { Table, Tr, Thumbnail, ImageCell, Th } from "./styles";

export default ({ users, setSelectedUser, setIsOpen }) => {
  return (
    <Table>
      <thead>
        <tr>
          <Th>Photo</Th>
          <Th>Name</Th>
          <Th>Username</Th>
          <Th>email</Th>
        </tr>
      </thead>
      <tbody>
        {users.map(({ name, picture, login, email }, index) => (
          <Tr
            key={`${name.first}-${index}`}
            onClick={() => {
              setSelectedUser(users[index]);
              setIsOpen(true);
            }}
          >
            <ImageCell>
              <Thumbnail src={picture.thumbnail} />
            </ImageCell>
            <td>
              <p>{`${name.first} ${name.last}`}</p>
            </td>
            <td>
              <p>{login.username}</p>
            </td>
            <td>
              <p>{email}</p>
            </td>
          </Tr>
        ))}
      </tbody>
    </Table>
  );
};

// (
//   <Ol>
//     {users.map(({ name, picture, login, email }, index) => (
//       <Li
//         key={`${name.first}-${index}`}
//         onClick={() => {
//           setSelectedUser(users[index]);
//           setIsOpen(true);
//         }}
//       >
//         <Thumbnail src={picture.thumbnail} />
//         <p>{`${name.first} ${name.last}`}</p>
//         <p>{login.username}</p>
//         <p>{email}</p>
//       </Li>
//     ))}
//   </Ol>
// )
