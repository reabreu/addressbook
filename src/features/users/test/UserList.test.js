import React from "react";
import { render } from "@testing-library/react";
import UserList from "../UserList";
import { usersData } from "./mock/data";

describe("<UserList /> spec", () => {
  it("Matches snapshot", () => {
    const { asFragment } = render(<UserList users={usersData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
