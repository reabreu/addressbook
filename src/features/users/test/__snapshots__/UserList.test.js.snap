// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList /> spec Matches snapshot 1`] = `
<DocumentFragment>
  <table
    class="sc-AxjAm eKQlwl"
  >
    <thead>
      <tr>
        <th
          class="sc-AxirZ kTvQwZ"
        >
          Photo
        </th>
        <th
          class="sc-AxirZ kTvQwZ"
        >
          Name
        </th>
        <th
          class="sc-AxirZ kTvQwZ"
        >
          Username
        </th>
        <th
          class="sc-AxirZ kTvQwZ"
        >
          email
        </th>
        <th
          class="sc-AxirZ kTvQwZ"
        />
      </tr>
    </thead>
    <tbody>
      <tr
        class="sc-AxiKw dUhbjF"
      >
        <td
          class="sc-AxhCb fRQPFI"
        >
          <img
            class="sc-AxhUy jBCOZP"
            src="https://randomuser.me/api/portraits/thumb/women/88.jpg"
          />
        </td>
        <td>
          <p>
            Leah Woods
          </p>
        </td>
        <td>
          <p>
            angryostrich988
          </p>
        </td>
        <td>
          <p>
            leah.woods@example.com
          </p>
        </td>
        <td>
          <svg
            class="sc-AxmLO dLqXOZ"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.516 3c2.382 0 4.487 1.564 4.487 4.712 0 4.963-6.528 8.297-10.003 11.935-3.475-3.638-10.002-6.971-10.002-11.934 0-3.055 2.008-4.713 4.487-4.713 3.18 0 4.846 3.644 5.515 5.312.667-1.666 2.333-5.312 5.516-5.312zm0-2c-2.174 0-4.346 1.062-5.516 3.419-1.17-2.357-3.342-3.419-5.515-3.419-3.403 0-6.484 2.39-6.484 6.689 0 7.27 9.903 10.938 11.999 15.311 2.096-4.373 12-8.041 12-15.311 0-4.586-3.414-6.689-6.484-6.689z"
            />
          </svg>
        </td>
      </tr>
      <tr
        class="sc-AxiKw dUhbjF"
      >
        <td
          class="sc-AxhCb fRQPFI"
        >
          <img
            class="sc-AxhUy jBCOZP"
            src="https://randomuser.me/api/portraits/thumb/women/77.jpg"
          />
        </td>
        <td>
          <p>
            Julia Jimenez
          </p>
        </td>
        <td>
          <p>
            happyostrich517
          </p>
        </td>
        <td>
          <p>
            julia.jimenez@example.com
          </p>
        </td>
        <td>
          <svg
            class="sc-AxmLO dLqXOZ"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.516 3c2.382 0 4.487 1.564 4.487 4.712 0 4.963-6.528 8.297-10.003 11.935-3.475-3.638-10.002-6.971-10.002-11.934 0-3.055 2.008-4.713 4.487-4.713 3.18 0 4.846 3.644 5.515 5.312.667-1.666 2.333-5.312 5.516-5.312zm0-2c-2.174 0-4.346 1.062-5.516 3.419-1.17-2.357-3.342-3.419-5.515-3.419-3.403 0-6.484 2.39-6.484 6.689 0 7.27 9.903 10.938 11.999 15.311 2.096-4.373 12-8.041 12-15.311 0-4.586-3.414-6.689-6.484-6.689z"
            />
          </svg>
        </td>
      </tr>
      <tr
        class="sc-AxiKw dUhbjF"
      >
        <td
          class="sc-AxhCb fRQPFI"
        >
          <img
            class="sc-AxhUy jBCOZP"
            src="https://randomuser.me/api/portraits/thumb/women/82.jpg"
          />
        </td>
        <td>
          <p>
            Sophia Johnston
          </p>
        </td>
        <td>
          <p>
            heavyostrich778
          </p>
        </td>
        <td>
          <p>
            sophia.johnston@example.com
          </p>
        </td>
        <td>
          <svg
            class="sc-AxmLO dLqXOZ"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.516 3c2.382 0 4.487 1.564 4.487 4.712 0 4.963-6.528 8.297-10.003 11.935-3.475-3.638-10.002-6.971-10.002-11.934 0-3.055 2.008-4.713 4.487-4.713 3.18 0 4.846 3.644 5.515 5.312.667-1.666 2.333-5.312 5.516-5.312zm0-2c-2.174 0-4.346 1.062-5.516 3.419-1.17-2.357-3.342-3.419-5.515-3.419-3.403 0-6.484 2.39-6.484 6.689 0 7.27 9.903 10.938 11.999 15.311 2.096-4.373 12-8.041 12-15.311 0-4.586-3.414-6.689-6.484-6.689z"
            />
          </svg>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
